dll/win32/shell32/utils.cpp

```diff
@@ -99,3 +99,151 @@ SHFindComputer(LPCITEMIDLIST pidlRoot, LPCITEMIDLIST pidlSavedSearch)
 
     return SUCCEEDED(hr);
 }
+
+static HRESULT Int64ToStr(LONGLONG llValue, LPWSTR pszValue, UINT cchValue)
+{
+    WCHAR szBuff[40];
+    UINT ich = 0, ichValue;
+#if (WINVER >= _WIN32_WINNT_VISTA)
+    BOOL bMinus = (llValue < 0);
+
+    if (bMinus)
+        llValue = -llValue;
+#endif
+
+    if (cchValue <= 0)
+        return E_FAIL;
+
+    do
+    {
+        szBuff[ich++] = (WCHAR)(L'0' + (llValue % 10));
+        llValue /= 10;
+    } while (llValue != 0 && ich < _countof(szBuff) - 1);
+
+#if (WINVER >= _WIN32_WINNT_VISTA)
+    if (bMinus && ich < _countof(szBuff))
+        szBuff[ich++] = '-';
+#endif
+
+    for (ichValue = 0; ich > 0 && ichValue < cchValue; ++ichValue)
+    {
+        --ich;
+        pszValue[ichValue] = szBuff[ich];
+    }
+
+    if (ichValue >= cchValue)
+    {
+        pszValue[cchValue - 1] = UNICODE_NULL;
+        return E_FAIL;
+    }
+
+    pszValue[ichValue] = UNICODE_NULL;
+    return S_OK;
+}
+
+static VOID
+Int64GetNumFormat(
+    _Out_ NUMBERFMTW *pDest,
+    _In_opt_ const NUMBERFMTW *pSrc,
+    _In_ DWORD dwNumberFlags,
+    _Out_writes_(cchDecimal) LPWSTR pszDecimal,
+    _In_ INT cchDecimal,
```

Comment:
Technically should be unsigned int (UINT), but yes, I know that GetLocaleInfoW takes the buffer size in (signed) INT, for whatever reason...